radio/src/gui/colorlcd/view_main.cpp

```diff
@@ -257,12 +257,12 @@ bool ViewMain::onTouchEnd(coord_t x, coord_t y)
 void ViewMain::onEvent(event_t event)
 {
   switch (event) {
-    case EVT_KEY_BREAK(KEY_MODEL):
+    case EVT_KEY_LONG(KEY_MODEL):
```

Comment:
I find it MUUUUUCCCCCCCHHHHH more intuitive this way

with just break it will act only if you release the key, even if you press it for ages to happen ... it was kind of the first thing I noticed with etx and immediately concluded it is flawed, because that's what one gets from "cheap" key implementations ...

as said somewhere, if you wanted to have it short, then you should catch it for BOTH break and long

so, whatever the goal, is, as it is currently it is bad :)

btw, note, that the change wasn't then consistent, since for TELEM it still was long ...

Reply:
the cleanest for handling short presses would actually be to add an event short or click which fires before long...

Reply:
> btw, note, that the change wasn't then consistent, since for TELEM it still was long ...

true, but at the same time, it is not the kind of action you use very often, once setup. And it can be accessed from the main menu as well.

Reply:
I was hoping to use the short press on TELEM for something really related to telemetry.

Reply:
well, I can understand why you/some want to have it rather short than long ... it's however against my genes which ask for "symmetry", "consistency", "logic" or however one wants to call it to have one button working differently for no current reason ...

if you think most will prefer short, I'm happy to change to this, but I think you would then want to have all buttons behave that way (at least for as long as there is nothing happening on a short TELEM)

Reply:
Ok, so let's have it "short", and give the TELEM short press a real function, as the screen setup has nothing to do with "telemetry", right?

Btw. when implementing "short press", how do you want to allow short and long being two different functions if you don't wait for the release? That wouldn't work, would it?

Reply:
> can someone pl test on a Tx16S?

By the way, you should have a look at your discord PMs ;-)

Reply:
> Btw. when implementing "short press", how do you want to allow short and long being two different functions if you don't wait for the release? That wouldn't work, would it?

short would fire before long
now it's FIRST, LONG, REPEAT, REPEAT, ..., BREAK, with short it would be FIRST, SHORT, LONG, REPEAT, REPEAT, ..., BREAK, that's all

for long you are not waiting for break even now, the "release" is achieved by calling killEvents(), would be very much so also for short
it would be actually bloody easy (if I don't miss a crucial point) to add short ....

Reply:
probably one doesn't even have to add short, first might do it, I'll try ...

Reply:
yeah, first is doing it ... :)

Reply:
this was not what I was trying to imply ... I just meant that this is how it works with the long